iothub/device/src/Transport/Amqp/IDeviceIdentity.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.Azure.Devices.Client.Transport.Amqp
+{
+    /// <summary>
+    /// Interface for device configurations and common attributes.
+    /// </summary>
+    internal interface IDeviceIdentity
+    {
+        /// <summary>
+        /// Authentication model for the device.
+        /// </summary>
+        AuthenticationModel AuthenticationModel { get; }
+
+        /// <summary>
+        /// AMQP transport layer settings of the device.
+        /// </summary>
+        AmqpTransportSettings AmqpTransportSettings { get; }
+
+        /// <summary>
+        /// Device connection string details.
+        /// </summary>
+        IotHubConnectionString IotHubConnectionString { get; }
+
+        /// <summary>
+        /// Device details and information.
```

Comment:
This is environment (OS), .NET version info, and client library version info, right? Is it really info about the device?

Reply:
updating.